### ISSUE TITLE
Error running globus-connect-server-io-setup

### ISSUE THREAD

**Issue by William-Hill** [ESGF/esgf-installer#443]
When running the command ```globus-connect-server-io-setup -c /etc/globus-connect-server-esgf.conf -v```, the following error occurs:

ENTER: GCMU.configure_trust_roots()
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/globus/connect/server/io/setup.py", line 138, in <module>
    ioobj.setup(reset=reset)
  File "/usr/lib/python2.6/site-packages/globus/connect/server/io/__init__.py", line 68, in setup
    self.configure_trust_roots(**kwargs)
  File "/usr/lib/python2.6/site-packages/globus/connect/server/io/__init__.py", line 303, in configure_trust_roots
    super(IO, self).configure_trust_roots(**kwargs)
  File "/usr/lib/python2.6/site-packages/globus/connect/server/__init__.py", line 470, in configure_trust_roots
    myproxy_ca_signing_policy)
  File "/usr/lib/python2.6/site-packages/globus/connect/security/__init__.py", line 121, in install_ca
    ca_hash = get_certificate_hash_from_data(ca_cert)
  File "/usr/lib/python2.6/site-packages/globus/connect/security/__init__.py", line 180, in get_certificate_hash_from_data
    cert_data + "\n" + err)
Exception: Error 1 getting certificate subject from /var/lib/globus-connect-server/myproxy-ca/cacert.pem
unable to load certificate
140135419197096:error:0906D06C:PEM routines:PEM_read_bio:no start line:pem_lib.c:708:Expecting: TRUSTED CERTIFICATE

**Comment by William-Hill** [comment]
I think I finally solved the Globus issue.  It appears this bug has been in the logic of the 2.x code but fails
silently in the background (ugh).  The problem occurs when invoking the following command: globus-connect-server-io-setup -c /etc/globus-connect-server-esgf.conf -v

The globus-connect-server-esgf.conf is a custom ESGF config file that gets created during the GridFTP setup in 
esg-globus script (GridFTP setup functions have been ext
racted to gridftp.py in ESGF 3.0).  This file is created in INI Style
and has a section called MyProxy. The globus-connect-server-io-setup script has a function location at /usr/lib/python2.6/site-packages/globus/connect/server/configfile.py
called get_myproxy_server() that gets called during the server setup (specifically in /usr/lib/python2.6/site-packages/globus/connect/server/__init__.py).

get_myproxy_server() checks to see if there is a MyProxy section and an option named Server globus-connect-server-esgf.conf. If so, it assumes returns true
and the script assumes that MyProxy will be setup.  So it then uses a function get_myproxy_ca() to check if there's an option called CA in globus-connect-server-esgf.conf.

def get_myproxy_ca(self):
        myproxy_ca = True
        if self.has_option(
                    ConfigFile.MYPROXY_SECTION,
                    ConfigFile.CA_OPTION):
            myproxy_ca = self.getboolean(
                    ConfigFile.MYPROXY_SECTION,
                    ConfigFile.CA_OPTION)
            if myproxy_ca == '':
                myproxy_ca = True
        return myproxy_ca

The problem lies in here because the function defaults to True if there's no option for CA in globus-connect-server-esgf.con. The esg-globus script
doesn't set a CA option, so the function returns True. This causes a problem because the following code exists in /usr/lib/python2.6/site-packages/globus/connect/server/__init__.py:

if self.conf.get_myproxy_ca():
                    myproxy_ca_dir = self.conf.get_myproxy_ca_directory()
                    myproxy_ca_cert = os.path.join(myproxy_ca_dir, "cacert.pem")

By get_myproxy_ca() defaulting to True, this causes the globus init script to look for the cacert.pem file that doesn't exist.
I've checked functional data-only nodes and the myproxy_ca_dir (defaults to /var/lib/globus-connect-server/myproxy-ca) doesn't exist.

This error fails silently in the Bash code.  But it raises an exception that kills execution in ESGF 3.0.

The solution is I've come up with is to only put the MyProxy section in the globus-connect-server-esgf.conf if IDP is part of 
the node type (i.e. not in a data-only node that setups up GridFTP).
